### PR TITLE
Fix getEnchantmentLevel

### PR DESCRIPTION
**Pull request by Intybyte**
# Description
Fix getEnchantmentLevel and hashCode

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.

### CODE REVIEW

**Comment by 4everTheOne** [comment]
Hi @Intybyte , thank you for your first contribution!

Could you add some unit tests to cover this methods?

File: src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java [hunk 1]
```diff
@@ -297,4 +309,9 @@ private static void handleLegacyEnchantmentsForDeserialization(@NotNull Map<Stri
 		}
 	}
 
+	@Override
+	public int hashCode() {
+		return Objects.hash(type, amount, itemMeta, durability);
+	}
```
Comment (4everTheOne) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java]: We should also override equals method

Comment (Intybyte) [reply src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java]: after a bit of talk on the discord  #1058 will cover that part, as it did so way beter than my case, i think i will revert the hashCode change for better merging later on

File: src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java [hunk 2]
```diff
@@ -184,6 +184,17 @@ public void addUnsafeEnchantment(@NotNull Enchantment ench, int level)
 		}
 	}
 
+	@Override
+	public int getEnchantmentLevel(Enchantment ench)
+	{
+		Preconditions.checkArgument(ench != null, "Enchantment cannot be null");
+
+		final ItemMeta meta = this.getItemMeta();
+		Preconditions.checkNotNull(meta, "Meta must not be null");
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java:193]: I think the paper implementation just defaults to 0 if there's no meta available. Although that's the case, we have a policy to throw exceptions whenever something is used improperly, so I think this is fine.

File: src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java [hunk 3]
```diff
@@ -184,6 +184,17 @@ public void addUnsafeEnchantment(@NotNull Enchantment ench, int level)
 		}
 	}
 
+	@Override
+	public int getEnchantmentLevel(Enchantment ench)
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java:188]: Although there's little to test here, could you just add a test that triggers this method, and checks if it does anything? (in ItemStackMockTest)

**Review by Thorinwasher — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java (hunk 3 above)
Comment (Intybyte) [reply src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java:188]: Alright

**Review by Thorinwasher — APPROVED** [review]

**Pull request merged**